### ISSUE TITLE
Products showing on frontend web interface only when the amount of products to show is changed

### ISSUE THREAD

**Issue by NotAndD** [magento/magento2#8216]
### Preconditions

1. Install Magento 2.1.3 with sample data.
2. Set up Rest integration.

### Steps to reproduce

1. Either using Rest API or the web backend interface, create a new product with name `Bag test creation`, setting it up correctly and put it inside category `Bag`.
2. Check in index management that everything is ready and up to date.
3. Go on the frontend website and open category `Bag`, browsing the products of the category.

### Expected result

1. The new product, normally showing up in the list of products.

### Actual result

1. The new product doesn't show up in the list of products
2. When the amount of products to show is changed (from 9 to another value), then the product is there with the other products. Not only that, I noticed that even other products were not showing out before and were showing out after changing the amounts to show.

Take a look at the products when amount to show is 15:
![magento_img1](https://cloud.githubusercontent.com/assets/14574992/22152570/d0d28302-df23-11e6-96c3-cd7931a15e05.jpg)

Then at the two pages of products when the amount to show is set to 9:
![magento_img2](https://cloud.githubusercontent.com/assets/14574992/22152572/d5565bce-df23-11e6-8134-e12116d6c592.jpg)

![magento_img3](https://cloud.githubusercontent.com/assets/14574992/22152576/d7ff9408-df23-11e6-8d52-cb9e02b82054.jpg)

As you can see, several products (the new one is one of them) are not showing as other products instead are repeated twice.

Interestingly enough, the number of total products show in the two case is the same, 14 products. Some id problems? Or am I missing something obvious?

**Comment by miakusha** [comment]
Hi @ziliquas, We cannot reproduce this issue.
Please provide Rest API payload. You used default Built-in Cache or Varnish?

**Comment by NotAndD** [comment]
The magento site where I tried use Built-in Cache.

It seems I am unable to reproduce this issue too. I don't know why, when I first found out the issue and took the screenshots and made tests, I had this issue.. I even take the time to count the items and make screenshots an all.. but now the issue is gone and I seem unable to reproduce it at all.

I suppose the problem had something to do with indexing and caching, mixed with rest apis heavy use on saving and updating elements but I can't help more unfortunately.
